### PR TITLE
Update GreatEncounters.groovy

### PR DESCRIPTION
**Pull request by Grayahm**
Implemented:
- Moonlight Stadium
- Felicity's Drawing
- Rare Candy (copied from EX Sandstorm)
- Baltoy
- Claydol

### CODE REVIEW

File: src/tcgwars/logic/impl/gen4/GreatEncounters.groovy [hunk 1]
```diff
@@ -1393,17 +1400,16 @@ public enum GreatEncounters implements LogicCardInfo {
           move "Psychic Balance", {
             text "If you have less cards in your hand than your opponent, draw cards until you have the same number of cards as your opponent. (If you have more or the same number of cards in your hand as your opponent, this attack does nothing.)"
             energyCost C
-            attackRequirement {}
             onAttack {
-              damage 0
```
Comment (luongthomas) [inline src/tcgwars/logic/impl/gen4/GreatEncounters.groovy:1398]: I haven't tested it yet but I wonder what the engine does when you do `draw -1`

**Comment by luongthomas** [comment]
While you're in the file, could you also update line 118 for MOONLIGHT_STADIUM_100 to add the `STADIUM` subType?

File: src/tcgwars/logic/impl/gen4/GreatEncounters.groovy (hunk 1 above)
Comment (Grayahm) [reply src/tcgwars/logic/impl/gen4/GreatEncounters.groovy:1398]: How so?

Comment (luongthomas) [reply src/tcgwars/logic/impl/gen4/GreatEncounters.groovy:1398]: The case when your hand size is larger than your opponent's hand size

File: src/tcgwars/logic/impl/gen4/GreatEncounters.groovy [hunk 2]
```diff
@@ -469,8 +469,15 @@ public enum GreatEncounters implements LogicCardInfo {
           pokePower "Cosmic Power", {
             text "Once during your turn , you may choose up to 2 cards from your hand and put them on the bottom of your deck in any order. If you do, draw cards until you have 6 cards in you hand. This power can’t be used if Claydol is affected by a Special Condition."
             actionA {
-            }
-          }
+              checkNoSPC()
+              assert my.hand.notEmpty : "You have no cards in your hand."
```
Comment (luongthomas) [inline src/tcgwars/logic/impl/gen4/GreatEncounters.groovy:473]: `checkLastTurn()` could also be added here too

Comment (Grayahm) [reply src/tcgwars/logic/impl/gen4/GreatEncounters.groovy:474]: Fixed

File: src/tcgwars/logic/impl/gen4/GreatEncounters.groovy [hunk 3]
```diff
@@ -2166,8 +2173,15 @@ public enum GreatEncounters implements LogicCardInfo {
         return basicTrainer (this) {
           text "You can play only one Supporter card each turn. When you play this card, put it next to your Active Pokémon. When your turn ends, discard this card.\nDiscard up to 2 cards from your hand. If you discard 1 card, draw 3 cards. If you discard 2 cards, draw 4 cards."
           onPlay {
-          }
-          playRequirement{
+            assert my.deck : "Your deck is empty"
+            assert my.hand.size() > 1 : "You can't play this card"
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen4/GreatEncounters.groovy:2177]: Should these assertions be playRequirements instead?

Comment (Grayahm) [reply src/tcgwars/logic/impl/gen4/GreatEncounters.groovy:2177]: Yes!

File: src/tcgwars/logic/impl/gen4/GreatEncounters.groovy (hunk 1 above)
Comment (Grayahm) [reply src/tcgwars/logic/impl/gen4/GreatEncounters.groovy:1398]: It's being used like this elsewhere on Holon Scientist and Lillie so I think it's okay

File: src/tcgwars/logic/impl/gen4/GreatEncounters.groovy (hunk 3 above)
Comment (Grayahm) [reply src/tcgwars/logic/impl/gen4/GreatEncounters.groovy:2177]: Fixed

**Pull request merged**